middleware/rules/rules-middleware.js

```diff
@@ -0,0 +1,63 @@
+// TODO: IMPLEMENT CRUD CALLS FROM API DOCUMENTATION FOR MIDDLEWARE.
+const request = require('request')
+
+// * SETTINGS FOR REQUEST PACKAGE BELOW
+const json = JSON.stringify({
+  client_id: process.env.client_id,
+  client_secret: process.env.client_secret,
+  audience: 'https://sauti-africa-market-prices.auth0.com/api/v2/',
+  grant_type: 'client_credentials'
+})
+
+const options = {
+  method: 'POST',
+  url: 'https://sauti-africa-market-prices.auth0.com/oauth/token',
+  headers: { 'content-type': 'application/json' },
+  body: json
+}
+
+// * RETRIEVE TOKEN FOR API MANAGEMENT &
+// * DYNAMIC USER GET REQUEST FOR USERS REQUESTING API KEY
+module.exports = async (req, res, next) => {
+  let apiToken
+
+  // * SET TOKEN FOR AUTHORIZED API USAGE
+  request(options, async (error, response, body) => {
+    apiToken = JSON.parse(body).access_token
+
+    // ! LOG TOKEN DATA BELOW
+    // console.log(apiToken)
```

Comment:
for testing or for devs? "no console logs in final code" but I don't hate commented out console logs with explanations in certain situations.